internal/pfcpsim/server.go

```diff
@@ -237,6 +232,51 @@ func (P pfcpSimService) CreateSession(ctx context.Context, request *pb.CreateSes
 				Build(),
 		}
 
+		// create as many PDRs as the number of app filters provided through pfcpctl
+		if len(request.AppFilters) > 0 {
+			pdrs = []*ieLib.IE{} // reset pdr slice
+			ID := uint16(i)
+
+			for _, appFilter := range request.AppFilters {
+				SDFFilter, gateStatus, err = parseAppFilter(appFilter)
+				if err != nil {
+					return &pb.Response{}, status.Error(codes.Aborted, err.Error())
+				}
+
+				log.Infof("Successfully parsed application filter. SDF Filter: %v", SDFFilter)
+
+				uplinkPDR := session.NewPDRBuilder().
+					WithID(ID).
+					WithMethod(session.Create).
+					WithTEID(uplinkTEID).
+					WithFARID(uplinkFarID).
+					AddQERID(sessQerID).
+					AddQERID(uplinkAppQerID).
```

Comment:
We should be using the same ID (ID, FAR ID...) for each uplink PDR of different applications. Same for downlink PDR.

Reply:
FARs shouldn't be shared between PDRs.